### ISSUE TITLE
The results are not the same when I set model to /home/sa/cuifengyu/detect/category/config/yolo11x-cls.yaml versus /home/sa/cuifengyu/detect/detection_models/ultralytics-main/weights/yolo11x-cls.pt in the config.yaml.

### ISSUE THREAD

**Issue by banjamn** [ultralytics/ultralytics#22704]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

When I train a classification model using YOLO11x-cls, I received different results when I set the model is "model: /home/sa/cuifengyu/detect/category/config/yolo11x-cls.yaml" or "model:/home/sa/cuifengyu/detect/detection_models/ultralytics-main/weights/yolo11x-cls.pt" in the config.yaml.
The other setting is the same.

### Additional

task: classify
mode: train
model: /home/sa/cuifengyu/detect/category/config/yolo11x-cls.yaml
pretrained: /home/sa/cuifengyu/detect/detection_models/ultralytics-main/weights/yolo11x-cls.pt
data: /home/sa/cuifengyu/detect/category/data/data_1031/split_data
epochs: 100

task: classify
mode: train
model: /home/sa/cuifengyu/detect/detection_models/ultralytics-main/weights/yolo11x-cls.pt
pretrained: /home/sa/cuifengyu/detect/detection_models/ultralytics-main/weights/yolo11x-cls.pt
data: /home/sa/cuifengyu/detect/category/data/data_1031/split_data
epochs: 100

when model is "model: /home/sa/cuifengyu/detect/category/config/yolo11x-cls.yaml"
 Epoch    GPU_mem       loss  Instances       Size
      1/100      7.73G      5.048         12        384: 100%|██████████| 505/505 [01:04<00:00,  7.79it/s]
               classes   top1_acc   top5_acc: 100%|██████████| 62/62 [00:04<00:00, 13.26it/s]
                   all     0.0292      0.128

when model is "model:/home/sa/cuifengyu/detect/detection_models/ultralytics-main/weights/yolo11x-cls.pt"
 Epoch    GPU_mem       loss  Instances       Size
      1/100      7.72G      4.081         12        384: 100%|██████████| 505/505 [01:04<00:00,  7.81it/s]
               classes   top1_acc   top5_acc: 100%|██████████| 62/62 [00:04<00:00, 12.70it/s]
                   all      0.505      0.818

**Comment by UltralyticsAssistant** [comment]
👋 Hello @banjamn, thank you for your detailed report and for using Ultralytics YOLO11 🚀  
This is an automated response to help you move forward quickly while an Ultralytics engineer reviews your issue and follows up soon 👨‍💻👩‍💻

From your description, it looks like you are seeing different classification results depending on whether you set `model` to a YAML config or directly to a `.pt` weight file in `config.yaml`. To help us understand whether this is a configuration issue or a potential bug, please provide a bit more detail.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) that includes:
- A minimal `config.yaml` showing both model configurations you used
- Exact command(s) you run to start training
- Ultralytics version (`yolo version` output) and your environment details (OS, CUDA, GPU)
- A very small sample of your dataset or a public dummy dataset that reproduces the difference
- Full training logs for at least the first few epochs for both runs

If this is a custom training ❓ Question about expected behavior when using YAML vs `.pt` model definitions, please also include:
- The contents of `/home/sa/cuifengyu/detect/category/config/yolo11x-cls.yaml`
- How the `.pt` file `/home/sa/cuifengyu/detect/detection_models/ultralytics-main/weights/yolo11x-cls.pt` was obtained (downloaded, trained, or exported)
- Any custom modifications you may have made to the model or config

In the meantime, you may find it helpful to review the [Docs](https://docs.ultralytics.com/) where you can see many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, including classification tasks, and where many common configuration questions are already answered.

Join the Ultralytics community where it suits you best:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧  
- For more in-depth discussions, check out [Discourse](https://community.ultralytics.com/)  
- Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community

## Upgrade

Before we dig deeper, please ensure you have upgraded to the latest `ultralytics` package and are using a supported environment, as many issues are already resolved in recent releases. Upgrade along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by banjamn** [comment]
Is it because I don't load the pretrained weights correctly? 


task: classify
mode: train
model: /home/sa/cuifengyu/detect/category/config/yolo11x-cls.yaml
pretrained: /home/sa/cuifengyu/detect/detection_models/ultralytics-main/weights/yolo11x-cls.pt
data: /home/sa/cuifengyu/detect/category/data/data_1031/split_data
epochs: 100
time: null
patience: 30
batch: 32
imgsz: 384
save: true
save_period: 10
cache: false
device: null
workers: 8
project: /home/sa/cuifengyu/detect/category/bird_classification/
name: train24_1112_deep_sp
exist_ok: false
pretrained: true

**Comment by banjamn** [comment]
I’ve already resolved the issue — it was because I didn’t load the pretrained weights.
The correct way to load the pretrained file is:

model = YOLO(model_name, task="classify", verbose=True)
model = model.load("./weights/yolo11x-cls.pt")

**Comment by glenn-jocher** [comment]
Good catch, and you’re exactly right that the difference came from not actually loading the pretrained weights when using the YAML. In the config you shared earlier, the second `pretrained: true` line overrides the path you set above, so training starts from scratch; to mirror your working Python flow, keep `model: /path/to/yolo11x-cls.yaml` and a single `pretrained: /path/to/yolo11x-cls.pt` (no duplicate key), which matches the pattern shown in the [image classification docs](https://docs.ultralytics.com/tasks/classify/).
